src/corelibs/U2Formats/src/FastaFormat.cpp

```diff
@@ -298,192 +286,173 @@ static void load(IOAdapter *io, const U2DbiRef &dbiRef, const QVariantMap &fs, Q
-    }
+static void writeHeaderToFile(IOAdapterWriter &writer, const QString &sequenceName, U2OpStatus &os) {
+    writer.write(os, FastaFormat::FASTA_HEADER_START_SYMBOL + sequenceName + '\n');
 }
 
-#define SAVE_LINE_LEN 1024 * 1024
+// TODO: a recommended width for FASTA lines is 80 chars to be readable in text editors.
+#define SAVE_LINE_LEN qint64(1024 * 1024)
 
-static void saveSequence(IOAdapter *io, const U2SequenceObject *sequence, U2OpStatus &os) {
-    writeHeaderToFile(io, sequence->getSequenceName(), os);
+static void saveSequenceObject(IOAdapterWriter &writer, const U2SequenceObject *sequence, U2OpStatus &os) {
+    writeHeaderToFile(writer, sequence->getSequenceName(), os);
     CHECK_OP(os, );
-
-    const int len = sequence->getSequenceLength();
-    for (int i = 0; i < len; i += SAVE_LINE_LEN) {
-        int chunkSize = qMin(SAVE_LINE_LEN, len - i);
-        const QByteArray chunkContent = sequence->getSequenceData(U2Region(i, chunkSize), os);
+    qint64 sequenceLength = sequence->getSequenceLength();
+    for (int i = 0; i < sequenceLength; i += SAVE_LINE_LEN) {
+        qint64 chunkSize = qMin(SAVE_LINE_LEN, sequenceLength - i);
+        QByteArray chunkContent = sequence->getSequenceData(U2Region(i, chunkSize), os);
         CHECK_OP(os, );
-        writeBlockToFile(io, chunkContent.constData(), chunkSize, os);
+        writer.write(os, QString::fromLatin1(chunkContent));
         CHECK_OP(os, );
     }
+    writer.write(os, "\n");
 }
 
-static void saveSequence(IOAdapter *io, const DNASequence &sequence, U2OpStatus &os) {
-    writeHeaderToFile(io, sequence.getName(), os);
+static void saveSequence(IOAdapterWriter &writer, const DNASequence &sequence, U2OpStatus &os) {
+    writeHeaderToFile(writer, sequence.getName(), os);
     CHECK_OP(os, );
 
-    const char *seq = sequence.seq.constData();
-    const int len = sequence.seq.length();
-    for (int i = 0; i < len; i += SAVE_LINE_LEN) {
-        const int chunkSize = qMin(SAVE_LINE_LEN, len - i);
-        writeBlockToFile(io, seq + i, chunkSize, os);
+    const char *seq = sequence.constData();
+    qint64 len = sequence.length();
+    for (qint64 i = 0; i < len; i += SAVE_LINE_LEN) {
+        int chunkSize = (int)qMin(SAVE_LINE_LEN, len - i);
+        writer.write(os, QString::fromLatin1(seq + i, chunkSize));
         CHECK_OP(os, );
     }
+    writer.write(os, "\n");
 }
 
-void FastaFormat::storeDocument(Document *doc, IOAdapter *io, U2OpStatus &os) {
-    //TODO: check saved op states!!!
-    foreach (GObject *o, doc->getObjects()) {
-        U2SequenceObject *seqObj = dynamic_cast<U2SequenceObject *>(o);
-        if (NULL != seqObj) {
-            saveSequence(io, seqObj, os);
+void FastaFormat::storeTextDocument(IOAdapterWriter &writer, Document *document, U2OpStatus &os) {
+    QList<GObject *> objects = document->getObjects();
+    for (GObject *object : qAsConst(objects)) {
+        if (auto sequenceObject = dynamic_cast<U2SequenceObject *>(object)) {
```

Comment:
Initializes in `if` statement is available since C++17. As far as I know we support C++11 only.

Reply:
We already have a bunch of places with C++17 : initializers in ifs(), shortened collections {a, b, c}, etc.

The answer is: if all our CI servers can compile the code -  the code is valid.